subprojects/packagefiles/tree-sitter/meson.build

```diff
@@ -0,0 +1,28 @@
+project('tree-sitter', 'c'
+  version: '0.15.7',
+  license: 'MIT',
+)
+
+tree_sitter = library('tree-sitter', [
+  'lib/src/lib.c',
+  ],
```

Comment:
Source files don't need to be wrapped in arrays, the parser expands them into varargs, see https://mesonbuild.com/Syntax.html#argument-flattening

Internally, it just sees `library('tree-sitter', 'lib/src/lib.c', include_directories: ..., ...)`